terraform/gitlab/gitlab.tf.json.template.py

```diff
@@ -1681,11 +1686,30 @@ def qq(*words):
                                         }
                                     }
                                 }
-                            }, indent=4)
+                                # FIXME: Hitting user_data size limits
+                                #        https://github.com/DataBiosphere/azul/issues/5314
+                            })
                         },
                     ],
+                    'power_state': {
+                        'mode': 'reboot',
+                        'message': 'Rebooting in order to enable FIPS mode',
+                        'condition': [
+                            'sysctl', 'crypto.fips_enabled', '|',
+                            'grep', '-oP', r'"(?<=crypto.fips_enabled = )\d+"'
+                        ]
```

Comment:
I have my doubt that this does what you want it to do.

When `condition` is a command, the exit status of that command is used. The regex matches any number after the equals sign and sysctl will always print a line. So both grep and sysctl will always succeed. I ran this on GL anvilprod and got this:

```
$ (sysctl crypto.fips_enabled | grep -oP "(?<=crypto.fips_enabled = )\d+") && echo yes
0
yes
```

FIPS is off but the command succeeds. It would also succeed if FIPS was on. So I am not sure this is even needed. How did you determine the need for (and test the efficacy of) this part?

Reply:
>How did you determine the need for (and test the efficacy of) this part?

The need was based on the request "… we just need to make sure we don't enter a loop." in the description of this ticket. I assumed this part would surely prevent that from happening, I didn't fully read the exit code of the command bit, and jumped to the `0: condition met, 1: condition not met` and since the command was printing a 0 or a 1, I equivocally assumed that the output was creating the conditions.

As you've pointed the lack of need for this. I was not appreciating that the `cloud-init: power_state` was running once per instance, and concluded that the instance was not rebooting on a loop due to the command in `condition` printing a 1 after the first reboot, which proved the efficacy (it being an incorrect conclusion).